### PR TITLE
GEOMESA-2182 Extend PartitionScheme Object to Java API

### PR DESCRIPTION
**Pull request by aheyne**
Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.fs.storage.api;
```
Comment (elahrvivaz) [inline geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java]: can we move this class to `org.locationtech.geomesa.fs.storage.common.interop` ? to keep packages separate per module and keep with our interop package naming conventions

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public static PartitionScheme extractFromSft(SimpleFeatureType sft) {
+        return org.locationtech.geomesa.fs.storage.common.PartitionScheme.extractFromSft(sft);
+    }
+
+    public static PartitionScheme apply(SimpleFeatureType sft, Map<String, Serializable> dsParams) {
+        return org.locationtech.geomesa.fs.storage.common.PartitionScheme.apply(sft, dsParams);
+    }
+
+    // TODO
+    //public static PartitionScheme apply(SimpleFeatureType sft, String pName, Map<String, String> opts)
```
Comment (elahrvivaz) [inline geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java]: remove commented out code

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java (hunk 1 above)
Comment (aheyne) [reply geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java]: Renamed

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java (hunk 2 above)
Comment (aheyne) [reply geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionSchemeUtils.java]: removed

**Pull request merged**